WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -262,6 +262,64 @@ def edit
     render :edit
   end
 
+  def payment_setup
+    @competition = Competition.find(params[:id])
+
+    client = create_oauth_client
+    oauth_params = {
+      scope: 'read_write',
+      state: @competition.id,
+    }
+    @authorise_url = client.auth_code.authorize_url(oauth_params)
+  end
+
+  def revoke_stripe_access
+    @competition = Competition.find(params[:id])
+    if @competition.connected_stripe_account_id
+      Stripe.api_key = ENVied.STRIPE_API_KEY
+      account = Stripe::Account.retrieve(@competition.connected_stripe_account_id)
+      account.deauthorize(
+        ENVied.STRIPE_CLIENT_ID,
+        { stripe_user_id: @competition.connected_stripe_account_id},
+      )
+      @competition.connected_stripe_account_id = nil
+      @competition.save
```

Comment:
We should a `save!` here, or check the return value of `.save` to see if it succeeded.